### ISSUE TITLE
Error when compileSdkVersion is set to non-literal

### ISSUE THREAD

**Issue by 1ec5** [bitrise-steplib/steps-install-missing-android-tools#5]
The Mapbox Android SDK project’s build.gradle file sets `compileSdkVersion` to [`rootProject.ext.compileSdkVersion`](https://github.com/mapbox/mapbox-gl-native/blob/87be62f9e4bbf990b63a1f4bed3939276cfd6a92/platform/android/MapboxGLAndroidSDK/build.gradle#L23), which is a reference to [an external dependencies.gradle file](https://github.com/mapbox/mapbox-gl-native/blob/87be62f9e4bbf990b63a1f4bed3939276cfd6a92/platform/android/dependencies.gradle#L4) that sets `compileSdkVersion` to a literal integer. We do this because we need to keep in sync multiple Gradle files for the SDK itself and some demo applications based on it. In some other cases, we have references to a gradle.properties file, such as `Integer.parseInt(project.ANDROID_BUILD_SDK_VERSION)`.

[This line](https://github.com/bitrise-steplib/steps-install-missing-android-tools/blob/a87902796d2d281206a87027270b116f2ba1af4b/analyzer/analyzer.go#L107) finds the `compileSdkVersion` in build.gradle but tries to parse its value as a literal integer, resulting in an error like this:

```
Failed to parse build.gradle at: /bitrise/src/platform/android/MapboxGLAndroidSDK/build.gradle, error: failed to determine compileSDKVersion, error: failed to parse compileSdkVersion (Integer.parseInt(project.ANDROID_BUILD_SDK_VERSION)), error: Malformed version: Integer.parseInt(project.ANDROID_BUILD_SDK_VERSION)
```

/cc @incanus @tobrun

**Comment by viktorbenei** [comment]
Hi @1ec5 ,

Thanks for the question/report!

So, the issue is that we tried to find a better solution to get the Build Tools, Android and Support Lib versions, but so far we couldn't find anything other than trying to parse these infos manually.

E.g. `gradle dependencies` does not list these base dependency versions, nor any other `gradle` command we found.

If you have a better solution or know a command we could use, please let us know! Otherwise, gradle can auto install these if you provide the license files (see: http://devcenter.bitrise.io/android/frequent-android-issues/#alternative-solution-for-the-license-error ), or you can install these manually with a Script step like:

```
echo y | android update sdk --no-ui --all --filter extra-android-m2repository | grep 'package installed'
```

But of course it would be way better if this step could reliably auto detect the config for you, and do this for you, without the need for licenses or the `android update sdk` command, so if you have any other idea how we could get these version numbers, please let us know (or send a PR ;) )!

**Comment by koral--** [comment]
I used something like that:
`printf "\nif (plugins.hasPlugin('com.android.application')) println 'compile sdk version: '+android.compileSdkVersion" | tee -a $(find . -name build.gradle) >/dev/null && ./gradlew -q buildEnvironment 2>/dev/null; git reset --hard >/dev/null`

However, when using Android Gradle plugin `2.2.0` or newer such step is not needed because missing SDK components will be automatically installed (assuming that license is accepted).

BTW `android` command is not functional in current SDK tools.

**Comment by godrei** [comment]
Hi @1ec5 ,

we released a new version ([v1.0.0](https://github.com/bitrise-steplib/steps-install-missing-android-tools/releases/tag/1.0.0)), it uses the solution mentioned by @koral-- . 

@koral-- thanks for the infos, i adopted your solution in the step and it seems it works great! You rock!

I'll close this issue, but if you need more info/help just let us know and we'll reopen it!
